interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: Yarn is a function that we can use to test our code to see if it will work as it should. Also, yarn can be used to create an app that we can run on our local server so we can see the app and how it changes as we begin to write our code.
 
-Researched answer:
+Researched answer: Yarn is a package manger that installs a piece of code from a global registry onto the local computer. Yarn was designed to have clean output logs for easy understandability. Yarn stores dependencies locally. Yarn generates a yarn.lock file. 
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function is when you need a function to run but you don't give put anything in it. This is done by using an empty (). 
 
-Researched answer:
+Researched answer: An anonymous function is when a function has been declared, however the function doesn't have a name. Anonymous functions can have multiple arguments, but only one expression. It is not accessable after its initial creation.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are consistent, they don't change. Props aka (properties) can change. For example, if you have a cup of water, the state of that substance will always be water. However, you can change the propety of that water if you freeze it it will become ice, and if you boil it it will become steam... but the state of that substance will always remain water.
 
-Researched answer:
+Researched answer: The state is controlled by the component, while props are controlled by whatever renders the comonent. Props will get data from an external source. States manage the internal data.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: DOM is basically our visual experience of the code. Everything we see on the screen, our mouse cursor scrolling across the page, clicking on something.... these are all examples of DOM.
 
-Researched answer:
+Researched answer: Document Object Model (DOM) defines how the browser reads your document and will creating a visual representation of what it reads.
 
 6. STRETCH: Which is the difference between a div and a span?
 
-Your answer:
+Your answer: A div is used to create dividers within a code. By basically diving the code into sections, it makes the code easier to work with and understand. This is helpful for things such as easier readabilty, accessabilty, styling, etc. 
 
-Researched answer:
+Researched answer: A div is a generic block level element used to divide a page into blocks. Mostly used for organizational purposes. A span is an inline element used to styling without dividing things into blocks. For example, if you had a the word "Mississippi" and you wanted to change the color of the text for only a few of the letters you would simply use a span element and place the letters that you want your styling to apply to within the span.
```

Comment:
Really great researched answers in this section. Keep working on that vocab for your initial answers.